graylog2-server/src/main/java/org/graylog/security/rest/EntitySharesResource.java

```diff
@@ -146,7 +147,7 @@ public EntityShareResponse prepareGenericShare(@ApiParam(name = "JSON Body") Pre
                 return entitySharesService.prepareShare(request.dependentEntityGRNs(), getCurrentUser());
             }
         } else {
-            return entitySharesService.prepareShare(getCurrentUser());
+            return entitySharesService.prepareShare(getCurrentUser(), Optional.ofNullable(request.selectedGranteeCapabilities()));
```

Comment:
Please avoid `Optional` as a method parameter. Especially for collection objects where the default can be empty.

```suggestion
            return entitySharesService.prepareShare(getCurrentUser(), Objects.requireNonNullElse(request.selectedGranteeCapabilities(), Map.of()));
```